### PR TITLE
Srp scroll

### PR DESCRIPTION
**Pull request by alias01**
Revamp of Srp page

- srp card
- infinite scroll
- filter added

### CODE REVIEW

File: src/components/molecules/MoleculeSRPCard.vue [hunk 1]
```diff
@@ -1,59 +1,74 @@
...
+    <div class="custom-card" @click="onDetails">
+        <div class="card-primary">
+            <div>
+                <div class="card-title">
+                    <p>
+                        {{ spot.Name }}
+                    </p>
+                </div>
+
+                <!-- <AtomRating
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:11]: please remove unused codes.

File: src/components/molecules/MoleculeSRPCard.vue [hunk 2]
```diff
@@ -1,59 +1,74 @@
...
-        </AtomButton>
+        </AtomButton> -->
     </div>
 </template>
 
 <script>
-import AtomRating from '../atoms/AtomRating.vue';
-import AtomParagraph from '../atoms/AtomParagraph.vue';
-import AtomButton from '../atoms/AtomButton.vue';
+// import AtomRating from '../atoms/AtomRating.vue';
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:64]: unused imports.

File: src/components/molecules/MoleculeSRPCard.vue [hunk 3]
```diff
@@ -76,121 +91,138 @@ export default {
...
+            background: var(--secondary-color);
+            right: 0;
+            bottom: 25%;
+            width: 2px;
+            height: 50%;
+        }
+    }
+
+    .card-primary {
+        display: flex;
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:132]: nit: keep attributes alphabetically sorted.

File: src/components/molecules/MoleculeSRPCard.vue [hunk 4]
```diff
@@ -76,121 +91,138 @@ export default {
...
+    }
+
+    .card-location {
+        align-items: center;
+        display: flex;
+        font-size: 12px;
+        gap: 5px;
+        position: relative;
+        line-height: 1.1;
+        color: #6b6b6b;
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:160]: why a new color code.
Isn't this defined as a a variable like other colors

File: src/components/molecules/MoleculeSRPCard.vue [hunk 5]
```diff
@@ -76,121 +91,138 @@ export default {
...
+        font-weight: var(--bold-font);
+    }
+
+    .card-spot-green {
+        color: green;
+        font-weight: var(--bold-font);
+    }
+
+    .card-btn {
+        // grid-column: 1 / 7;
```
Comment (arrogant-ant) [inline src/components/molecules/MoleculeSRPCard.vue:223]: unused codes

File: src/components/templates/TemplateSrp.vue [hunk 6]
```diff
@@ -15,36 +46,39 @@
...
                 :key="reRender"
             ></MapContainer>
-            <SearchInput class="map-search" @changed="onChange"></SearchInput>
         </div>
     </div>
 </template>
 
 <script>
-import PaginationBody from '../extras/PaginationBody.vue';
+// import PaginationBody from '../extras/PaginationBody.vue';
```
Comment (arrogant-ant) [inline src/components/templates/TemplateSrp.vue:61]: unused

File: src/store/map/index.js [hunk 7]
```diff
@@ -133,13 +134,17 @@ const mutations = {
...
+        state.filteredSrpResults = srpResults;
+    },
 };
 
 const actions = {
     async searchLocation({ commit }, query) {
         const token =
             'pk.eyJ1IjoiaWFtZmlhc2NvIiwiYSI6ImNrOWZiankzdjA5d2kzbWp3NGNzNmIwaHAifQ.E2UwYdvpjc6yNoCmBjfTaQ';
-        const url = `/geocoding/v5/mapbox.places/${query}.json?access_token=${token}&proximity=77.4977,12.9716`;
+        const url = `/geocoding/v5/mapbox.places/${query}.json?access_token=${token}&worldview=in&country=in`;
```
Comment (arrogant-ant) [inline src/store/map/index.js:147]: tested this?

File: src/store/map/index.js [hunk 8]
```diff
@@ -179,6 +185,25 @@ const actions = {
         }
         return state.recentSearch;
     },
+
+    updateSrpResults({ commit }, filterOptions) {
+        let filteredSrpResults = [];
+        if (filterOptions.length < 2 && filterOptions.length !== 0) {
```
Comment (arrogant-ant) [inline src/store/map/index.js:191]: can be simplified to  
```filterOptions.length == 1```

File: src/views/PageSrp.vue [hunk 9]
```diff
@@ -48,13 +49,17 @@ export default {
             deep: true,
             immediate: true,
         },
+        filteredSrpResults() {
```
Comment (arrogant-ant) [inline src/views/PageSrp.vue:52]: please add some comments

**Review by arrogant-ant — CHANGES_REQUESTED** [review]

File: src/components/molecules/MoleculeSRPCard.vue (hunk 4 above)
Comment (alias01) [reply src/components/molecules/MoleculeSRPCard.vue:160]: this is some other hexcode ...not defined in variable.

File: src/components/molecules/MoleculeSRPCard.vue (hunk 3 above)
Comment (alias01) [reply src/components/molecules/MoleculeSRPCard.vue:132]: not following the alphabetic order..
below is the order of attributes
            'position',
            'top',
            'right',
            'bottom',
            'left',
            'z-index',
            'display',
            'justify-content',
            'align-items',
            'float',
            'clear',
            'overflow',
            'overflow-x',
            'overflow-y',
            'margin',
            'margin-top',
            'margin-right',
            'margin-bottom',
            'margin-left',
            'padding',
            'padding-top',
            'padding-right',
            'padding-bottom',
            'padding-left',
            'width',
            'min-width',
            'max-width',
            'height',
            'min-height',
            'max-height',
            'font-size',
            'font-family',
            'font-weight',
            'border',
            'border-style',
            'border-width',
            'border-color',
            'border-top',
            'border-top-style',
            'border-top-width',
            'border-top-color',
            'border-right',
            'border-right-style',
            'border-right-width',
            'border-right-color',
            'border-bottom',
            'border-bottom-style',
            'border-bottom-width',
            'border-bottom-color',
            'border-left',
            'border-left-style',
            'border-left-width',
            'border-left-color',
            'border-radius',
            'text-align',
            'text-justify',
            'text-indent',
            'text-overflow',
            'text-decoration',
            'white-space',
            'color',
            'background',
            'background-position',
            'background-repeat',
            'background-size',
            'background-color',
            'background-clip',
            'opacity',
            'filter',
            'list-style',
            'outline',
            'visibility',
            'box-shadow',
            'text-shadow',
            'resize',
            'transition',

File: src/components/molecules/MoleculeSRPCard.vue [hunk 10]
```diff
@@ -76,121 +91,138 @@ export default {
...
+            word-wrap: break-word;
+        }
+    }
+
+    .card-rating {
+        margin-bottom: unset;
+    }
+
+    .card-distance {
+        background-color: rgb(214, 253, 255);
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/molecules/MoleculeSRPCard.vue:182]: no... its a lighter version for blend in with white.

File: src/components/templates/TemplateSrp.vue [hunk 11]
```diff
@@ -59,8 +93,15 @@ export default {
             type: Number,
         },
     },
-    emits: ['changed', 'flyToSrp', 'details'],
-
+    data() {
+        return {
+            filterOptions: ['Available', 'Rented out'],
+            isFilter: false,
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateSrp.vue:99]: not now

File: src/components/templates/TemplateSrp.vue [hunk 12]
```diff
@@ -74,48 +115,155 @@ export default {
         selected(spot) {
             console.log(spot);
```
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateSrp.vue:116]: not now

File: src/components/templates/TemplateSrp.vue (hunk 11 above)
[the remark this answers is not in the document]
Comment (alias01) [reply src/components/templates/TemplateSrp.vue:99]: done

File: src/store/map/index.js (hunk 7 above)
Comment (alias01) [reply src/store/map/index.js:147]: proximity restricting searching in other places...now restricted to india...if typed correctly will show correct result

File: src/store/map/index.js (hunk 8 above)
Comment (alias01) [reply src/store/map/index.js:191]: done

File: src/components/molecules/MoleculeSRPCard.vue (hunk 3 above)
Comment (arrogant-ant) [reply src/components/molecules/MoleculeSRPCard.vue:120]: for where you got the list?

File: src/store/map/index.js (hunk 7 above)
Comment (arrogant-ant) [reply src/store/map/index.js:147]: What do you mean by "typed correctly"
Most of our searches is around bangalore (thanks to google ads configuration)
So will the fuzzy search quality improve or drop?

File: src/components/molecules/MoleculeSRPCard.vue (hunk 10 above)
[the remark this answers is not in the document]
Comment (arrogant-ant) [reply src/components/molecules/MoleculeSRPCard.vue]: these free floating colors hexcode will present difficulty when we plan to change color scheme

**Review by arrogant-ant — APPROVED** [review]

**Pull request merged**